lib/puppet/pops/adapters.rb

```diff
@@ -91,6 +91,7 @@ def extract_tree_text
       last = first + @adapted.length
       @adapted.eAllContents.each do |m|
         m_offset = m.offset
+        next if m_offset.nil?
```

Comment:
We do not have tests that all expressions gets offset and length set correctly. Needless to say that would be a substantial task as source snippets covering all possible paths through the grammar would need to be covered. You almost would need to generate them from the grammar itself to be certain. 
Coming back to this PR, there will be a problem unless there always is an empty string expression at the end of the interpolation. I can see either fixing the missing offset in this ticket, or creating a new ticket for that problem. (Should not be hard to fix, probably just one call to locate in the grammar).